### ISSUE TITLE
Improve column view autoscroll 

### ISSUE THREAD

**Issue by hanaral** [elementary/files#1608]
## Problem

Wanting to just to drag and drop an item from one directory to another has the entire view change, which not only hinders my ability to just move something from A to B, but in most cases only serves as a punishment for misclicking in a directory. 
This is made worse by the delay in the before the autoscroll kicks in as the user likely already moved the item over the destination folder.

## Proposal

Replace the click based autoscroll with one that only moves when items are moved to the edge like in a generic scroll view.

## Prior Art

Finder

**Comment by jeremypw** [comment]
I tend to agree the click-based autoscroll can be troublesome, but the idea is that if you click on a folder in a slot that is right on the edge of the view, then the newly created column column should not be invisible.  How would you deal with this?

**Comment by jeremypw** [comment]
Autoscroll also keeps the currently selected item in view when you are navigating left and right using the keyboard.  However, it does seem to autoscroll unnecessarily sometimes - think it is to keep at least some of the columns to each side of the currently focused one in view.

**Comment by hanaral** [comment]
I'm not sure about mouse, but I use a trackpad and my main way is to click and swipe. It's not tedious, and I much prefer being able to move the view to the position I need manually.
I still think the issue is just how much space the columns take up, I set my default column width to 120  - which fits 3.5 columns in a moderately-sized window.

**Comment by hanaral** [comment]
@jeremypw Just thought of something that could bridge the gap for users that rely on it, how about making the view move start instantly and smoothly; a bit like a relaxed glide 
Another idea would be to only do it on a doube click, but that would only be discorverable by accident and feels like not actually improving how it works.

**Comment by jeremypw** [comment]
Another possibility is to have an option to turn autoscroll off or lose it altogether.  I note that ListView does not autoscroll if an expanded folder goes off-screen.
